Assets/MRTK/SDK/Features/UX/Scripts/InteractiveElement/States/StateManager.cs

```diff
@@ -140,7 +141,8 @@ public InteractionState SetStateOn(string stateName)
 
             if (state != null)
             {
-                if (state.Value != 1)
+                // Only update the state value and invoke events if InteractiveElement is Active
+                if (state.Value != 1 && InteractiveElement.Active)
```

Comment:
Consider factoring into method like "IsStateValid" or "CanTransitionToState" to reduce duplicate code